File: tools-httpclient/src/main/java/cn/gjing/http/ConnectionFactory.java
package cn.gjing.http;

import javax.net.ssl.HttpsURLConnection;
import javax.net.ssl.SSLContext;
import javax.net.ssl.SSLSocketFactory;
import javax.net.ssl.TrustManager;
import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;

/**
 * Connection generation factory
 * @author Gjing
 **/
class ConnectionFactory {

    /**
     * request url
     */
    private String requestUrl;

    ConnectionFactory(String requestUrl) {
        this.requestUrl = requestUrl;
    }

    /**
     * get httpsConnection instance
     * @return HttpsURLConnection
     */
    HttpsURLConnection getHttps() {
        try {
            SSLContext sslContext = SSLContext.getInstance("SSL");
            TrustManager[] tm = {new HttpsManager()};
            sslContext.init(null, tm, new java.security.SecureRandom());
            SSLSocketFactory ssf = sslContext.getSocketFactory();
            URL url = new URL(this.requestUrl);
            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
            connection.setSSLSocketFactory(ssf);
            connection.setRequestProperty("Charset", "UTF-8");
            connection.setDoOutput(true);
            connection.setDoInput(true);
            connection.setConnectTimeout(5000);
            connection.setReadTimeout(10000);
            return connection;
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

    /**
     * get httpConnection instance
     * @return HttpURLConnection
     */
    HttpURLConnection getHttp() {
        try {
            URL url = new URL(this.requestUrl);
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setRequestProperty("Charset", "UTF-8");
            connection.setDoOutput(true);
            connection.setDoInput(true);
            connection.setConnectTimeout(5000);
            connection.setReadTimeout(10000);
            return connection;
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: tools-httpclient/src/main/java/cn/gjing/http/ConnectionFactory.scala
package cn.gjing.http

import java.io.IOException
import java.net.{HttpURLConnection, URL}
import java.security.SecureRandom

import javax.net.ssl.{HttpsURLConnection, SSLContext, TrustManager}

/**
 * @author Gjing
 **/
class ConnectionFactory(requestUrl:String) {
  /**
   * Get httpsConnection instance
   *
   * @return HttpsURLConnection
   */
  private[http] def getHttps = try {
    val sslContext = SSLContext.getInstance("SSL")
    val tm:Array[TrustManager] = Array(new HttpsManager)
    sslContext.init(null, tm, new SecureRandom)
    val ssf = sslContext.getSocketFactory
    val url = new URL(this.requestUrl)
    val connection = url.openConnection.asInstanceOf[HttpsURLConnection]
    connection.setSSLSocketFactory(ssf)
    connection.setRequestProperty("Charset", "UTF-8")
    connection.setDoOutput(true)
    connection.setDoInput(true)
    connection.setConnectTimeout(5000)
    connection.setReadTimeout(10000)
    connection
  } catch {
    case e: Exception =>
      e.printStackTrace()
      null
  }

  /**
   * Get httpConnection instance
   *
   * @return HttpURLConnection
   */
  private[http] def getHttp = try {
    val url = new URL(this.requestUrl)
    val connection = url.openConnection.asInstanceOf[HttpURLConnection]
    connection.setRequestProperty("Charset", "UTF-8")
    connection.setDoOutput(true)
    connection.setDoInput(true)
    connection.setConnectTimeout(5000)
    connection.setReadTimeout(10000)
    connection
  } catch {
    case e: IOException =>
      e.printStackTrace()
      null
  }
}
